test/e2e/upgrade_test.go

```diff
@@ -46,13 +46,14 @@ func upgradeTestSuite(t *testing.T) {
 	require.NoError(t, err, "error configuring upgrade")
 
 	// get current Windows node state
+	// TODO: waitForWindowsNodes currently loads nodes into global context, so we need this (even though BYOH
+	// 		 nodes are not being upgraded/tested here). Remove as part of https://issues.redhat.com/browse/WINC-620
 	err = testCtx.waitForWindowsNodes(gc.numberOfMachineNodes, false, true, false)
 	require.NoError(t, err, "wrong number of Machine controller nodes found")
 	err = testCtx.waitForWindowsNodes(gc.numberOfBYOHNodes, false, true, true)
 	require.NoError(t, err, "wrong number of ConfigMap controller nodes found")
 
 	t.Run("Operator version upgrade", testUpgradeVersion)
-	t.Run("Version annotation tampering", testTamperAnnotation)
```

Comment:
In this commit, this line + the deletion of the function are really the only changes that are accomplishing the goals you outlined in the commit message.
Everything you've done is beneficial for the codebase as a whole, but it would be better to separate these related small quality of life refactors into a separate commit, for the sake of reviewers. It makes it easier for us to see what should have a functional impact, and what shouldn't :)

Reply:
ACK, broke it out into separate commits.